### PR TITLE
Slim seasons data

### LINKED ISSUE #471: Slim down seasons data
[Calculating the season of items](https://github.com/DestinyItemManager/DIM/blob/e6ae335774a41a1243c0ac136c51183b66785f69/src/app/inventory/store/season.ts#L28) pulls in some very large JSON into the initial bundle. While there's work we could (and should) do to lazy load this info, it'd also be great if we could slim down that info significantly - for example, by not populating season data for items where we can get their season from the overlay.

### PR DESCRIPTION
**Pull request by delphiactual**
Closes #471 

~~still need to ensure eventWatermarks is generated before seasons.json~~ this is already handled by `prioritizedScripts` in main.js

```
What I'm suggesting is:

1. If an item definition contains a watermark, via displayVersionWatermarkIcons or iconWatermark or iconWatermarkShelved, and that watermark can be associated with a season, that hash should not be in seasons.json because we won't bother using it.
2. If the remaining items after step 1 do not include one of the sources in source-to-season.json, that source should be removed from source-to-season.json.
3. seasons_backup.json should be removed in favor of seasons.json
4. Rework the season-detection logic in DIM to look in watermark-to-season.json, then fall back to source-to-season.json, then fall back to a very trimmed-down seasons.json that only contains hashes that could ever get to this level of fallback.

```

- [x] 1
- [x] 2
- [x] 3
- [x] 4 Addressed by https://github.com/DestinyItemManager/DIM/pull/10369

### CODE REVIEW

**Comment by delphiactual** [comment]
not sure how to proceed on step 2, any and all assistance would be greatly appreciated

**Comment by bhollis** [comment]
Nice!

> If the remaining items after step 1 do not include one of the sources in source-to-season.json, that source should be removed from source-to-season.json.

I'm not sure, but I *think* you can move the code starting at `// loop through collectibles` on line 18 down under the code you just added, so that it only looks at the items that have already been filtered based on overlay.

File: output/source-to-season.json [hunk 1]
```diff
@@ -1,61 +1,44 @@
...
-    "186854335": 14,
-    "194661944": 1,
     "210885364": 13,
     "266896577": 5,
-    "277706045": 14,
-    "281362298": 20,
     "286427063": 12,
     "287889699": 13,
-    "288436121": 9,
+    "315474873": 1,
```
Comment (delphiactual) [inline output/source-to-season.json:19]: This is not a seasonally unique source
image

**Comment by delphiactual** [comment]
I am going to need some more assistance on this @bhollis @nev-r @robojumper

**Review by bhollis — APPROVED** [review]
Looks good!

**Pull request merged**